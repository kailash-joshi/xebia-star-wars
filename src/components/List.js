import React from 'react';
// import PropTypes from 'prop-types';
import './list.scss';
const List = ({planets}) => {
    const planetList = planets.map((planet)=>{
        return <div key={planet.name}><font size={planet.population}>{planet.name}</font></div>
    })
    return(
        <div className="listItem">{planetList}</div>
    )
}
List.defaultProps = {
    planets: []
}
export default List;
