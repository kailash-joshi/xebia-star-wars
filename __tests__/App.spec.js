import React from 'react';
import renderer from 'react-test-renderer';
import List from '../src/components/List';

describe('List', () => {
    const component = renderer.create(
        <List planets={new Array()} />
    );
    test('snapshot renders', () => {
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});