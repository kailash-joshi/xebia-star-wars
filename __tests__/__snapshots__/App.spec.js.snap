// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List snapshot renders 1`] = `
<div
  className="listItem"
/>
`;
